lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -123,10 +132,25 @@ def matches_arity?(actual, name)
           expectation.expect_unlimited_arguments = @unlimited_arguments
           expectation.expect_arbitrary_keywords  = @arbitrary_keywords
 
+          expectation
+        end
+
+        def matches_arity?(actual, name)
+          expectation = setup_method_signature_expectation
+
           return true if expectation.empty?
 
-          Support::StrictSignatureVerifier.new(method_signature_for(actual, name)).
-            with_expectation(expectation).valid?
+          begin
+            Support::StrictSignatureVerifier.new(method_signature_for(actual, name)).
+              with_expectation(expectation).valid?
+          rescue NameError
+            return true if @ignoring_method_signature_failure
+            raise ::ArgumentError, "The #{matcher_name} matcher requires that " \
```

Comment:
Does it have to be specified as a global const?

Reply:
You mean the `::` probably not